lib/portlayer/metrics/subscription.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/docker/docker/pkg/pubsub"
+
+	"github.com/vmware/govmomi/object"
+
+	"github.com/vmware/vic/lib/portlayer/exec"
+)
+
+// subscription is a 1:1 relationship to a container and a
+// 1:M relationship to subscribers
+type subscription struct {
+	c *exec.Container
+
+	pub                 *pubsub.Publisher
+	devices             object.VirtualDeviceList
+	deviceInstanceToKey map[string]string
+
+	diskNames    []string // container's virtualDisk names
+	networkNames []string // container's network names
+}
+
+// DeviceName will return the name associated with the metric instance id
+func (sub *subscription) DeviceName(instance string) string {
+	var name string
+
+	// did we previously find this device
+	if name, exists := sub.deviceInstanceToKey[instance]; exists {
+		return name
+	}
+
+	// convert instance to key - we are expecting regular failures, so no logging
+	// or returning of error
+	key, err := strconv.Atoi(instance)
```

Comment:
do you have an example of an `instance` that shouldn't be translated we can add to doc?
I'm not clear on what would be passed in that's not translated - I also cannot see where this method is called.